OpenRA.Mods.Common/Traits/Infantry/TakeCover.cs

```diff
@@ -12,12 +12,13 @@
 using System.Collections.Generic;
 using System.Linq;
 using OpenRA.Primitives;
+using OpenRA.Support;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("Make the unit go prone when under attack, in an attempt to reduce damage.")]
-	public class TakeCoverInfo : TurretedInfo
+	public class TakeCoverInfo : TurretedInfo, IObservesVariablesInfo, IRulesetLoaded
```

Comment:
You don't need `, IObservesVariablesInfo, IRulesetLoaded`. Those are inherited from ConditionalTrait.